Shopping List/ShoppingCollectionViewController.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  ShoppingCollectionViewController.swift
+//  Shopping List
+//
+//  Created by Cameron Dunn on 1/18/19.
+//  Copyright © 2019 Lambda School. All rights reserved.
+//
+
+import UIKit
+
+private let reuseIdentifier = "Cell"
+
+class ShoppingCollectionViewController: UICollectionViewController, ButtonDelegate {
+    func tappedButton(on: CustomCollectionViewCell) {
+        model.editItemInList(sender: on)
+        model.saveList()
+        collectionView?.reloadData()
+    }
+    let model = ShoppingListModel()
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if(segue.identifier == "NextSegue"){
+            model.createShoppingListArray()
+            let destination = segue.destination as! CheckoutViewController
+            destination.model = model
+        }
+    }
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        model.createShoppingList()
+        model.loadList()
+        
+        // Uncomment the following line to preserve selection between presentations
+        // self.clearsSelectionOnViewWillAppear = false
+
+        // Register cell classes
+        self.collectionView!.register(UICollectionViewCell.self, forCellWithReuseIdentifier: reuseIdentifier)
```

Comment:
If you register cell in storyboard, you don't need this line of code. This probably creates another cell, so you're ending with two cells per row. Check this [stackoverflow answer](https://stackoverflow.com/a/36000801/2916820)
```suggestion

```